src/main/mw.c

```diff
@@ -291,10 +291,11 @@ void mwDisarm(void)
 #define TELEMETRY_FUNCTION_MASK (FUNCTION_TELEMETRY_FRSKY | FUNCTION_TELEMETRY_HOTT | FUNCTION_TELEMETRY_SMARTPORT | FUNCTION_TELEMETRY_LTM)
 
 void releaseSharedTelemetryPorts(void) {
-    serialPort_t *sharedPort = findSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP);
-    while (sharedPort) {
-        mspReleasePortIfAllocated(sharedPort);
-        sharedPort = findNextSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP);
+    for(serialPort_t *sharedPort = findSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP);
+        sharedPort;
+        sharedPort = findNextSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP)
+        ) {
+        mspSerialReleasePortIfAllocated(sharedPort);
```

Comment:
@hydra , I'm with @ledvinap here, the use of the `for` loop is documentary. The iterator idiom
```
for (x_t *x = firstX(); x != null; x = nextX()) {...
```
is fairly common and indicates to the reader that something is being iterated over. In my opinion the use of a `while` loop obscures this iteration. (I prefer to keep the superfluous `!= null` comparison.)

I also agree that the serial port find and find next functions could do with some work - but that can be left to a future PR - let's keep focused on the task at hand.